source/R/steekproeftrekking_nabehandeling.R

```diff
@@ -64,16 +64,33 @@ thin_sample <- function(sample, thin_dist) {
 
 replace_by_existing <- function(sample,
                                 existing_points,
+                                id_existing_points,
+                                gebied,
                                 overlap_prop = 0.5,
                                 sbp_file) {
+  # Determine where HOL and OL is in gebied
+  ol <- selectie_openheid(
+    gebied = gebied,
+    ol_strata = c("OL"))
+
+  hol <- selectie_openheid(
+    gebied = gebied,
+    ol_strata = c("HOL"))
   # Recalculate sbp stratum existing points
   old_points <- existing_points %>%
+    st_transform(crs = 31370) %>%
```

Comment:
Indien existing_points afkomstig is van SOVON, dan moeten we hier mogelijk die speciale transformatie gebruiken? Het zou kunnen dat met deze transformatie de punten niet op de juiste plek komen waardoor de vervangprocedure de mist in gaat.
Bij voorkeur kan deze speciale transformatie (indien ze nodig is) buiten de targets pipeline gebeuren en het correct getransformeerde object opgeslagen worden als geojson of geopackage en dan hier ingelezen worden. 

Hier is code die we daarvoor gebruiken:
De CRS moet aangepast worden naar van Amersfoort naar Lambert 72. Dit doen we via een specifieke pipeline, zie: https://inbo.github.io/tutorials/tutorials/spatial_transform_crs_2/

```{r}
library(sf)
library(dplyr)
Sys.setenv("PROJ_NETWORK" = "ON")

pipelines <- sf_proj_pipelines("EPSG:28992", "EPSG:31370")
glimpse(pipelines)

# We select the pipeline with lowest accuracy (< 0.02), by filtering on accuracy.
# If the grids are installed, this should match the first line of the pipelines object.

chosen_pipeline_definition <- pipelines %>%
  filter(accuracy < 0.02) %>% 
  pull(definition)

mas_31370_pipeline <- 
  st_transform(st_geometry(mas_geojson_new), "EPSG:31370",
               pipeline = chosen_pipeline_definition) %>% 
  st_sf(st_drop_geometry(mas_geojson_new), geometry = .) %>% 
  as_tibble() %>% 
  st_as_sf()
```

Ik ping ook even @wlangera om al dan niet te bevestigen dat deze comment terecht is of niet.

Reply:
Correct! Ik bedacht me dit gisterenavond nog dat dit nog moest gebeuren.
Let op dat je `Sys.setenv("PROJ_NETWORK" = "ON")` ofwel runt vòòr je **sf** laadt ofwel er direct achter (zie [deze issue](https://github.com/r-spatial/sf/issues/2166)), maar ik denk dat het met jouw code wel zal werken.

Ik denk dat dit inderdaad buiten de pipeline moet gebeuren. De input van de functie is dan een geojson of geopackage met Lambert 72 coördinaten.

Reply:
Misschien in de begin van de functie een check toevoegen? `stop(sf::st_crs(existing_points) == 31370, call. = FALSE)`

Reply:
Bedoel je niet `stopifnot(sf::st_crs(existing_points)$input == "EPSG:31370")`, nadat de juiste transformatie is toegepast buiten de pipeline.

Reply:
Ja je hebt gelijk. Vooral om in de toekomst te vermijden dat we dit zouden vergeten.

Reply:
Dit bleek niet nodig want de shapefile was in WGS 84 CRS. Er is enkel een stopifnot toegevoegd die de pipeline tegenhoudt als we in de toekomst toch Amersfoort zouden tegenkomen.